### PR TITLE
[DOC] Array doc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Adds section 'In brief' for Array#[]=.

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
There may be too many examples in the new section;  suggest deletions please.

**Comment by BurdetteLamar** [comment]
I think the doc for Array#[]= may be missing the cases where the operand is an array;  if so, that will be for a different PR.

File: array.c [hunk 1]
```diff
@@ -2284,6 +2294,73 @@ ary_aset_by_rb_ary_splice(VALUE ary, long beg, long len, VALUE val)
  *
  *  Assigns elements in +self+; returns the given +object+.
  *
+ *  In brief:
```
Comment (peterzhu2118) [inline array.c:2297]: IMO the "in brief" section stops being useful when it tries to exhaustively cover all the cases. It becomes harder to read than the regular documentation.

Comment (BurdetteLamar) [reply array.c:2297]: Agree;  pared.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, will you be reviewing this?  I've pared down the examples (as I suggested in the OP might be necessary).

**Review by peterzhu2118 — APPROVED** [review]

**Comment by peterzhu2118** [comment]
Sorry, I must have missed this.

**Pull request merged**